### PR TITLE
test: bluetooth: resume_adv: Create advertising resume tester (Bsim) for issue #51256

### PR DESCRIPTION
**Pull request by eslam-cloud2nd**
The tester includes three examples run on `babblesim` to simulate `To Reproduce` like #50438
1- First Application is central_hr to simulate "NRF Connect"
2- Second Application is peripheral_hr 
3- Third Application is central_peripheral_hr

Signed-off-by: Esalm Essa <eslam@cloud2gnd.com>

### CODE REVIEW

File: tests/bluetooth/bsim_bt/bsim_test_resume_adv/central_hr/boards/rv32m1_vega_ri5cy.overlay [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+};
+
+&intmux0_ch3 {
+	status = "okay";
+};
+
+&generic_fsk {
+	interrupt-parent = <&intmux0_ch3>;
+	status = "okay";
+};
```
Comment (Thalley) [inline tests/bluetooth/bsim_bt/bsim_test_resume_adv/central_hr/boards/rv32m1_vega_ri5cy.overlay:22]: Why are we adding board overlays for boards that are not the `nrf52_bsim` board in the BSIM test directory?

File: tests/bluetooth/bsim_bt/bsim_test_resume_adv/peripheral_hr/sample.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    tags: bluetooth
+  sample.bluetooth.peripheral_hr_rv32m1_vega_ri5cy:
+    platform_allow: rv32m1_vega_ri5cy
+    tags: bluetooth
+    build_only: true
+  sample.bluetooth.peripheral_hr.frdm_kw41z_shield:
+    harness: bluetooth
+    depends_on: arduino_serial
+    tags: bluetooth
+    extra_args: SHIELD=frdm_kw41z
```
Comment (Thalley) [inline tests/bluetooth/bsim_bt/bsim_test_resume_adv/peripheral_hr/sample.yaml:17]: Why are we adding sample descriptions in the BSIM directory?

File: tests/bluetooth/bsim_bt/bsim_test_resume_adv/central_hr/README.rst [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+************
+
+* BlueZ running on the host, or
+* A board with BLE support
+
+Building and Running
+********************
+
+This sample can be found under :zephyr_file:`samples/bluetooth/central_hr` in the
+Zephyr tree.
```
Comment (Thalley) [inline tests/bluetooth/bsim_bt/bsim_test_resume_adv/central_hr/README.rst:23]: Incorrect location and also not a sample

**Review by Thalley — CHANGES_REQUESTED** [review]

**Review by aescolar — CHANGES_REQUESTED** [review]
Same review as https://github.com/zephyrproject-rtos/zephyr/pull/53134#pullrequestreview-1238562085

**Pull request closed without merging**